web/src/components/apps/Dashboard.jsx

```diff
@@ -94,58 +135,33 @@ class Dashboard extends Component {
     })
       .then(async (res) => {
         if (!res.ok) {
-          this.setState({ gettingAppLicenseErrMsg: body.error });
+          setGettingAppLicenseErrMsg(body.error);
           return;
         }
 
         const body = await res.json();
         if (body === null) {
-          this.setState({ appLicense: {}, gettingAppLicenseErrMsg: "" });
+          setAppLicense({});
+          setGettingAppLicenseErrMsg("");
         } else if (body.success) {
-          this.setState({
-            appLicense: body.license,
-            gettingAppLicenseErrMsg: "",
-          });
+          setAppLicense(body.license);
+          setGettingAppLicenseErrMsg("");
         } else if (body.error) {
-          this.setState({
-            appLicense: {},
-            gettingAppLicenseErrMsg: body.error,
-          });
+          setAppLicense({});
+          setGettingAppLicenseErrMsg(body.error);
```

Comment:
This is a common theme- basically every `setState` had to be replaced with a hook.